### PR TITLE
Add tests for some NumberFormat options edge cases

### PR DESCRIPTION
**Pull request by littledan**
These tests are regression tests for V8 failures at top of tree,
https://bugs.chromium.org/p/v8/issues/detail?id=6015
https://bugs.chromium.org/p/v8/issues/detail?id=6016

Neither issue is present in JSC or SpiderMonkey. They are fixed
by the V8 patch
https://codereview.chromium.org/2717613005

### CODE REVIEW

File: test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+description: >
+    When a currency is used in Intl.NumberFormat and minimumFractionDigits is
+    not provided, maximumFractionDigits should be range-checked against it.
+include: [assert.js]
+---*/
+
+assert.throws(RangeError,
+              () => new Intl.NumberFormat('en', {
+                  style: 'currency',
+                  currency: 'USD',
```
Comment (leobalter) [inline test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js]: I'm assuming __ISO 4217__ defines the currency digits to 2. It would be good to add another assertion in this file using some currency without a defined digits to get the default value 2 from [currencydigits](https://tc39.github.io/ecma402/#sec-currencydigits).

File: test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+                  maximumFractionDigits: 1
+              }));
```
Comment (leobalter) [inline test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js]: the indentation is weird, but not a blocker to land this PR

File: test/intl402/NumberFormat/fraction-digit-options-read-once.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    Regression test for https://bugs.chromium.org/p/v8/issues/detail?id=6015
+include: [assert.js]
+---*/
+
+var minCounter = 0;
+var maxCounter = 0;
+new Intl.NumberFormat("en", { get minimumFractionDigits() { minCounter++ },
+                              get maximumFractionDigits() { maxCounter++ } });
+assert.sameValue(1, minCounter);
+assert.sameValue(1, maxCounter);
```
Comment (leobalter) [inline test/intl402/NumberFormat/fraction-digit-options-read-once.js]: we can improve this a bit to get the order they are called as well:

```js
var calls = [];

new Intl.NumberFormat("en", { get minimumFractionDigits() { calls.push('minimumFractionDigits') },
                              get maximumFractionDigits() { calls.push('maximumFractionDigits') } });
assert.sameValue(calls.length, 2);
assert.sameValue(calls[0], 'minimumFractionDigits');
assert.sameValue(calls[1], 'maximumFractionDigits');
```

File: test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2017 the V8 project authors. All rights reserved.
```
Comment (leobalter) [inline test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js]: nit: this file can be renamed to `dft-currency-mnfd-range-check-mxfd.js`.

It should not block this PR as well.

File: test/intl402/NumberFormat/fraction-digit-options-read-once.js (hunk 3 above)
Comment (littledan) [inline test/intl402/NumberFormat/fraction-digit-options-read-once.js]: Thanks for the suggestion, done.

File: test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js (hunk 2 above)
Comment (littledan) [inline test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js]: Changed, do you like the new indentation better?

File: test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js (hunk 1 above)
Comment (littledan) [inline test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js]: Added CLF

File: test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+// This code is governed by the license found in the LICENSE file.
+
+/*---
+esid: ECMA-402 #sec-setnfdigitoptions
+description: >
+    When a currency is used in Intl.NumberFormat and minimumFractionDigits is
+    not provided, maximumFractionDigits should be range-checked against it.
+include: [assert.js]
```
Comment (littledan) [inline test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js]: Done

File: test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2017 the V8 project authors. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
+
+/*---
+esid: ECMA-402 #sec-setnfdigitoptions
```
Comment (littledan) [inline test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js]: I removed the #, but shouldn't we mention somehow that this is in 402, not 262?

File: test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js (hunk 4 above)
Comment (littledan) [inline test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js]: Done

File: test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js (hunk 2 above)
Comment (leobalter) [inline test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js]: thank you!

**Comment by leobalter** [comment]
this is merged at a9d9ab1. Thanks!

**Comment by leobalter** [comment]
I also removed e0b71ae from the changes as it belongs to #924, right?

File: test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js (hunk 6 above)
Comment (leobalter) [inline test/intl402/NumberFormat/default-currency-maximum-fraction-digits.js]: the current path is already saying that.

**Comment by littledan** [comment]
@leobalter Oh thanks for catching that, my mistake.
